app/src/main/java/com/kickstarter/viewmodels/CommentsViewHolderViewModel.kt

```diff
@@ -95,21 +100,30 @@ interface CommentsViewHolderViewModel {
         private val replyToComment = PublishSubject.create<Comment>()
         private val flagComment = PublishSubject.create<Comment>()
         private val viewCommentReplies = PublishSubject.create<Comment>()
+        private val isCommentEnableThreads = PublishSubject.create<Boolean>()
 
         val inputs: Inputs = this
         val outputs: Outputs = this
 
+        private val optimizely: ExperimentsClientType = environment.optimizely()
+
         init {
             this.commentInput
                 .map { it.comment }
                 .filter { ObjectUtils.isNotNull(it) }
                 .compose(bindToLifecycle())
-                .subscribe { this.commentCardStatus.onNext(cardStatus(it)) }
+                .subscribe {
+                    this.commentCardStatus.onNext(cardStatus(it))
+                }
 
             this.commentInput
                 .compose(Transformers.combineLatestPair(environment.currentUser().observable()))
                 .compose(bindToLifecycle())
-                .subscribe { this.isCommentActionGroupVisible.onNext(isActionGroupVisible(it.first, it.second)) }
+                .subscribe {
+                    this.isReplyButtonVisible.onNext(
+                        shouldReplyButtonBeVisible(it.first, it.second, optimizely.isFeatureEnabled(OptimizelyFeature.Key.COMMENT_ENABLE_THREADS))
```

Comment:
I think this is the only place that we should have logic about showing and hiding the reply button.

Reply:
Updated to add that the reply button is actually tied to the card status -> deleted and failed, so we need to update when the status is updated 👍🏽